### PR TITLE
CalculatorApp first pull request

### PR DESCRIPTION
**Pull request by Anvinalias**
I previously made a mistake creating a calculator instead of a math app, but I've reused the code and made necessary adjustments based on the Microsoft article. Suggest any changes if necessary.

Thanks for reviewing.

### CODE REVIEW

File: Calculator/Program.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                }
+
+                opt.GetOperands(userChoice);
+
+                Console.WriteLine();
+                Console.WriteLine("Press 'N' to quit or any other key to Continue");
+                continueChoice = Console.ReadLine();
+
+            } while (continueChoice != "n" && continueChoice != "N"); 
+
```
Comment (chrisjamiecarter) [inline Calculator/Program.cs:46]: Remove blank line.

File: CalculatorLibrary/CalculatorLibrary.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
+﻿
```
Comment (chrisjamiecarter) [inline CalculatorLibrary/CalculatorLibrary.cs:1]: Remove blank line.

File: CalculatorLibrary/CalculatorLibrary.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+namespace CalculatorLibrary
+{
+    public class Option
+    {
+        static Operation Op = new Operation();
+        static List<string> listItems = new List<string>();
+
+        JsonWriter writer;
+        public Option()
```
Comment (chrisjamiecarter) [inline CalculatorLibrary/CalculatorLibrary.cs:12]: Blank line needed here.

File: CalculatorLibrary/CalculatorLibrary.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        }
+
+        public void JsonFinish()
+        {
+            writer.WriteEndArray();
+            writer.WriteEndObject();
+            writer.Close();
+        }
+    }
+    internal class Operation
```
Comment (chrisjamiecarter) [inline CalculatorLibrary/CalculatorLibrary.cs:110]: Blank line needed here.

File: CalculatorLibrary/CalculatorLibrary.cs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            }
+
+            Console.WriteLine("Result history:");
+
+            foreach (string i in listItems)
+            {
+                Console.WriteLine(i);
+            }
+        }
+
```
Comment (chrisjamiecarter) [inline CalculatorLibrary/CalculatorLibrary.cs:152]: Remove blank line.

File: CalculatorLibrary/CalculatorLibrary.cs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
```
Comment (chrisjamiecarter) [inline CalculatorLibrary/CalculatorLibrary.cs:154]: Remove blank line.

**Review by chrisjamiecarter — APPROVED** [review]
Hey @Anvinalias ,

Great job on your calculator application! 🎉 You are making great progress.
Below are a few suggestions that I think will help you take this to the next level:

**Solution Structure**
Re-read the [Code Reviews](https://thecsharpacademy.com/article/52/code-reviews) academy page. Specifically step 3.
This will make it easier for code-reviewers to pull down and review your code in the future.
Your solution should be contained in a folder as per below:

```
// Template.
.
└── [ProjectName].[GitHubUserName]/
    ├── [SolutionDirectory1]/
    ├── [SolutionDirectory2]/
    └── [ProjectName].sln

// Your example.
.
└── Calculator.Anvinalias/
    ├── Calculator/
    ├── CalculatorLibrary/
    └── Calculator.sln
```

**Style Guide Compliance**
Your code is generally well-structured, but to align with best practices, it would be beneficial to follow [Microsoft's style guide](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions) more closely.
This will ensure consistency and make your code easier for others to read and maintain.
For example:
- Pay attention to the whitespace in your code. 
- Insert a single blank line between method declarations at the same indentation level (especially in CalculatorLibrary.cs line: 110).
- No random blank lines.
- No blank lines on cascading trailling }
```
            /// Code here.
        }
    }
}
```
- Use file-scoped namespaces to save an indentation level
```
namespace CalculatorLibrary
{
    // Code here.
}
```
vs
```
namespace CalculatorLibrary;
// Code here.
```

**User Experience**
While the current flow of your calculator functions works, there’s an opportunity to improve the user experience.
You could clear the console after each calculation, before you show the main menu. And utilise `Console.ReadKey()` to await a user input to continue after they have seen and read the calculation result.
This would allow the user to only view one whole 'transaction' in the console window at a time.

Feel free to reach out if you have any questions or if you'd like to collaborate on any of these points.

Well done on a fantastic job, and these are just a few suggestions to support your growth as a developer.
I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter

File: Calculator/Program.cs (hunk 1 above)
Comment (Anvinalias) [reply Calculator/Program.cs:46]: Thanks for the detailed reviews will work on all the suggestions.

**Pull request closed without merging**